server/odc-service/src/main/java/com/oceanbase/odc/service/structurecompare/StructureComparisonTask.java

```diff
@@ -0,0 +1,159 @@
+            @NonNull Long structureComparisonTaskId,
+            @NonNull User user,
+            @NonNull StructureComparisonTaskRepository structureComparisonTaskRepository,
+            @NonNull StructureComparisonTaskResultRepository structureComparisonTaskResultRepository,
+            @NonNull ObjectStorageFacade objectStorageFacade) {
+        this.srcConfig = srcConfig;
+        this.tgtConfig = tgtConfig;
+        initTaskResult(structureComparisonTaskId);
+        this.taskId = taskId;
+        this.user = user;
+        this.structureComparisonTaskResultRepository = structureComparisonTaskResultRepository;
+        this.structureComparisonTaskRepository = structureComparisonTaskRepository;
+        this.objectStorageFacade = objectStorageFacade;
+    }
+
+    private void initTaskResult(Long structureComparisonTaskId) {
+        this.taskResult.setTaskId(structureComparisonTaskId);
+        if (!srcConfig.getBlackListMap().isEmpty()) {
+            this.taskResult.setComparingList(srcConfig.getBlackListMap().get(DBObjectType.TABLE).stream()
+                    .map(tableName -> new Comparing(DBObjectType.TABLE, tableName))
+                    .collect(Collectors.toList()));
+        }
+    }
+
+    @Override
+    public DBStructureComparisonTaskResult call() throws Exception {
+        SecurityContextUtils.setCurrentUser(user);
+        StructureComparisonTraceContextHolder.trace(user.getId(), taskId);
+        try {
+            log.info("Structure comparison task starts, id={}", taskResult.getTaskId());
+            long startTimestamp = System.currentTimeMillis();
+            taskResult.setStatus(TaskStatus.RUNNING);
+            try {
+                List<DBObjectComparisonResult> results = comparator.compare(srcConfig, tgtConfig);
+                saveComparisonResult(results, tgtConfig.getConnectType().getDialectType());
+                taskResult.setStatus(TaskStatus.DONE);
+            } catch (Exception e) {
+                log.warn("Failed to run structure comparison task, id={}, error message={}", taskResult.getTaskId(),
+                        e.getMessage(), e);
+                taskResult.setStatus(TaskStatus.FAILED);
+            }
+            log.info("Structure comparison task ends, id={}, task status={}, time consuming={} seconds",
+                    taskResult.getTaskId(), taskResult.getStatus(),
+                    (System.currentTimeMillis() - startTimestamp) / 1000);
+            return taskResult;
+        } finally {
+            StructureComparisonTraceContextHolder.clear();
+        }
+    }
+
+    private void saveComparisonResult(List<DBObjectComparisonResult> results, DialectType dialectType) {
+        List<StructureComparisonTaskResultEntity> entities =
+                results.stream().map(result -> result.toEntity(taskResult.getTaskId(), dialectType)).collect(
+                        Collectors.toList());
+        structureComparisonTaskResultRepository.saveAll(entities);
```

Comment:
the default impl of 'saveAll' in jpa is not bulk insert, this may cause the performance problem when entities is too large

Reply:
On complex

Reply:
Optimized it